app/src/main/ui/src/hooks/useCasUser.tsx

```diff
@@ -0,0 +1,7 @@
+import {StateSetter, useAPI} from "./useAPI";
+import {CasUser, CasUserService} from "../services/CasUser";
+
+export const useCasUser = (): [CasUser, boolean, StateSetter<CasUser>] => {
+    const [user, isLoading, setUser] = useAPI<CasUser>(CasUserService.get)
+    return [user ?? {principal: "default", admin: false}, isLoading, setUser]
```

Comment:
Why `default`? Is this ever shown to the user? wouldnt an empty string be better?
no user => no username

Reply:
For local development it's better to show a default username instead of an empty string.